src/addIndex.js

```diff
@@ -20,7 +20,7 @@ var curryN = require('./curryN');
  * @category List
  * @sig ((a ... -> b) ... -> [a] -> *) -> (a ..., Int, [a] -> b) ... -> [a] -> *)
```

Comment:
sounds good. i also don't understand the intended meaning of the ellipsis after `a` in the function sig.